src/compiler/parser/index.js

```diff
@@ -410,6 +416,31 @@ function processAttrs (el) {
   }
 }
 
+function checkVBindingVal (name: string, value: string): void {
+  let func
+
+  // there may be template error, leave it to the compiler detectErrors
+  try {
+    func = new Function(`(${value})`)
+  } catch (ex) {
+    return
+  }
+
+  try {
+    func()
```

Comment:
En, this is just a check **in dev mode**. If the expression passed down is plain js code like 'alert(1)',  it'll be warned.

```html
<component :val="alert(1)"></component>
```